core/src/main/java/com/linecorp/armeria/internal/common/DefaultSplitHttpResponse.java

```diff
@@ -313,18 +298,12 @@ private void onNext0(HttpData httpData) {
          * Completes the specified trailers.
          */
         private void completeTrailers(HttpHeaders trailers) {
-            HeadersFuture<HttpHeaders> trailersFuture = DefaultSplitHttpResponse.this.trailersFuture;
-            if (trailersFuture != null) {
-                trailersFuture.doComplete(trailers);
+            if (response instanceof DecodedHttpResponse) {
+                // DecodedHttpResponse.trailers() is used instead.
                 return;
             }
```

Comment:
Maybe no need to do this at all because `doComplete()` will return silently anyway? Just some comment may be enough.